webapi/src/main/scala/org/knora/webapi/responders/v1/StandoffResponderV1.scala

```diff
@@ -0,0 +1,634 @@
+
+                _ = if (existingMappingResponse.statements.nonEmpty) {
+                    throw BadRequestException(s"mapping Iri $mappingIri already exists")
+                }
+
+                createNewMappingSparql = queries.sparql.v1.txt.createNewMapping(
+                    triplestore = settings.triplestoreType,
+                    dataNamedGraph = namedGraph,
+                    mappingIri = mappingIri,
+                    label = label,
+                    mappingElements = mappingElements
+                ).toString()
+
+                // Do the update.
+                createResourceResponse <- (storeManager ? SparqlUpdateRequest(createNewMappingSparql)).mapTo[SparqlUpdateResponse]
+
+                // check if the mapping has been created
+                newMappingResponse <- (storeManager ? SparqlConstructRequest(getExistingMappingSparql)).mapTo[SparqlConstructResponse]
+
+                _ = if (newMappingResponse.statements.isEmpty) {
+                    log.error(s"Attempted a SPARQL update to create a new resource, but it inserted no rows:\n\n$newMappingResponse")
+                    throw UpdateNotPerformedException(s"Resource $mappingIri was not created. Please report this as a possible bug.")
+                }
+
+                // get the mapping from the triplestore and cache it thereby
+                _ = getMappingFromTriplestore(mappingIri, userProfile)
+
+
+            } yield {
+                CreateMappingResponseV1(mappingIri = mappingIri, userdata = userProfile.userData)
+            }
+
+
+            createMappingFuture.recoverWith {
+                case validationException: SAXException => throw BadRequestException(s"the provided mapping is invalid: ${validationException.getMessage}")
+
+                case ioException: IOException => throw NotFoundException(s"The schema could not be found")
+
+                case unknown: Exception => throw BadRequestException(s"the provided mapping could not be handled correctly: ${unknown.getMessage}")
+            }
+
+        }
+
+        for {
+        // Don't allow anonymous users to create a mapping.
+            userIri: IRI <- Future {
+                userProfile.userData.user_id match {
+                    case Some(iri) => iri
+                    case None => throw ForbiddenException("Anonymous users aren't allowed to create mappings")
+                }
+            }
+
+            // check if the given project Iri represents an actual project
+            projectInfo: ProjectInfoResponseV1 <- (responderManager ? ProjectInfoByIRIGetRequestV1(
+                iri = projectIri,
+                Some(userProfile)
+            )).mapTo[ProjectInfoResponseV1]
+
+            knoraIdUtil = new KnoraIdUtil
+
+            // TODO: make sure that has sufficient permissions to create a mapping in the given project
```

Comment:
@subotic Does the new user and projects design allow any known user to add **new data**  to a project's named graph?